src/App.tsx

```diff
@@ -142,6 +142,19 @@ function App() {
   } = useSDK({
     onSessionChanged: handleSessionChangedNotification,
     onWalletNotify: handleNotification,
+    onWalletAnnounce: (ev: Event) => {
+      const customEvent = ev as CustomEvent;
+      const newExtensionId = customEvent.detail.params.extensionId ?? '';
+      const newEntry: WalletMapEntry = {
+        params: customEvent.detail.params,
+        eventName: customEvent.detail.params.uuid,
+      };
+      setExtensionId(newExtensionId);
+      setWalletMapEntries((prev) => ({
+        ...prev,
+        [customEvent.detail.params.uuid]: newEntry,
+      }));
+    },
   });
```

Comment:
```suggestion
    onWalletAnnounce: useCallback(
      (ev: Event) => {
        const customEvent = ev as CustomEvent;
        const newExtensionId = customEvent.detail.params.extensionId ?? '';
        const newEntry: WalletMapEntry = {
          params: customEvent.detail.params,
          eventName: customEvent.detail.params.uuid,
        };
        setExtensionId(newExtensionId);
        setWalletMapEntries((prev) => ({
          ...prev,
          [customEvent.detail.params.uuid]: newEntry,
        }));
      },
      [setExtensionId, setWalletMapEntries],
    ),
  });
```

Should we keep following the same pattern here of extracting this above logic to it's own separate handleWalletAnnounce function ?

Also, wrapping it in a useCallback with the dependencies [setExtensionId, setWalletMapEntries] would do good to avoid unnecessary re-renders downstream

Reply:
done here: https://github.com/MetaMask/test-dapp-multichain/pull/30/commits/da05648bd565b4b7a77e606e0665f073370bd2c2